config/v1/types_dns.go

```diff
@@ -96,3 +103,35 @@ type DNSList struct {
 
 	Items []DNS `json:"items"`
 }
+
+// DNSPlatformSpec holds cloud-provider-specific configuration
+// for DNS administration.
+// +union
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'AWS' ?  has(self.aws) : !has(self.aws)",message="aws configuration is required when platform is AWS, and forbidden otherwise"
+type DNSPlatformSpec struct {
+	// type is the underlying infrastructure provider for the cluster.
+	// Allowed values: "", "AWS".
+	// DNSPlatformType is a subset of Infrastructure PlatformType.
```

Comment:
It isn't clear what `DNSPlatformType` refers to.  Is this what you mean to say?  
```suggestion
	// The set of allowed values for this field is a subset of the allowed values for Infrastructure PlatformType.
```

Reply:
fixed (removed) -- leftover from a previous iteration